arelle/WebCache.py

```diff
@@ -712,40 +726,43 @@ def _downloadFile(
                                         messageCode="webCache:unsuccessfulRetrieval",
                                         messageArgs={"error": err, "URL": url},
                                         level=logging.ERROR)
-                    if os.path.exists(tempFilepath):
-                        os.remove(tempFilepath)
+                    if temporaryFilename is not None:
+                        temporaryFilename.unlink(missing_ok=True)
                     return False
 
             # rename temporarily named downloaded file to desired name
-            if os.path.exists(filepath):
+            with SoftFileLock(f'{filepath}.lock'):
```

Comment:
Add a timeout so we don't potentially hang forever if Arelle crashes on a previous execution?